## Project-Zomato API-I

##### BY: ADITI DONA
link: https://developers.zomato.com/api/v2.1/search?entity_id=1&entity_type=city&start=1&count=20 

In [24]:
import requests as req
import json

res=req.get('https://developers.zomato.com/api/v2.1/cuisines',params={'city_id':'1'},
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
for i in data['cuisines']:
    if i['cuisine']['cuisine_name']=='Mexican':
        print(i['cuisine']['cuisine_id'])

73


1) Fetch the entityid and entitytype of place Connaught Place.

2) Print the entityid and entitytype

In [36]:
import requests as req
import json
res=req.get('https://developers.zomato.com/api/v2.1/locations',params={'query':'Connaught Place'},
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
for i in data['location_suggestions']:
    print(i['entity_type'],i['entity_id'])


subzone 104


3) Fetch the top 10 best-rated restaurant serving ‘Mexican’ cuisine present in Connaught Place.

4) Print the restaurant name, user rating and restaurant id.

In [108]:
import requests as req
import json
val={'entity_id':104,'entity_type':'subzone','cuisines':'73','sort':'rating','order':'desc','count':'10'}
res=req.get('https://developers.zomato.com/api/v2.1/search',params=val,
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
for i in data['restaurants']:
    print(i['restaurant']['name'],end=" ")
    rat=i['restaurant']['user_rating']
    print(rat['aggregate_rating'],end=" ")
    print(i['restaurant']['id'])

Chili's Grill & Bar 4.6 18537536
38 Barracks 4.6 18241537
Ministry Of Beer 4.5 18418277
Xero Degrees 4.4 18729557
Farzi Cafe 4.3 18233593
Bunta Bar Live 4.3 18623406
Unplugged Courtyard 4.2 18216901
Excuse Me Boss 4.2 307036
Punjabi By Nature 4.2 306497
The Art House Bar n Cafe 4.2 18863766


5) Fetch the category id of category type ‘Cafes’ using Zomato API.

6) Print the category_id.

In [134]:
import requests as req
import json
res=req.get('https://developers.zomato.com/api/v2.1/categories',
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
for i in data['categories']:
    if i['categories']['name']=='Cafes':
        print(i['categories']['id'])

6


7) Fetch the best-rated restaurant for ‘Mexican’ cuisine with category type ‘ Cafes’ present in Connaught Place using Zomato API.

8) Print the restaurant name, user rating and restaurant id.

In [107]:
import requests as req
import json
val={'entity_id':104,'entity_type':'subzone','cuisines':'73','category':'6','sort':'rating','order':'desc','count':'10'}
res=req.get('https://developers.zomato.com/api/v2.1/search',params=val,
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
for i in data['restaurants']:
    print(i['restaurant']['name'],end=" ")
    rat=i['restaurant']['user_rating']
    print(rat['aggregate_rating'],end=" ")
    print(i['restaurant']['id'])

Xero Degrees 4.4 18729557
The Art House Bar n Cafe 4.2 18863766
Cafe Delhi Heights 4.1 18579363
Cafe Coffee Day - The Square 3.8 303034


9) Fetch the latest review of the best-rated restaurant for ‘Mexican’ cuisine with Category type ‘Cafes’ present in Connaught Place 

10) using Zomato API.

11) Print the name of user ,user rating and review text.

import requests as req
import json
val={'res_id':'18729557'}
res=req.get('https://developers.zomato.com/api/v2.1/reviews',params=val,
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
data
for i in data['user_reviews']:
    print(i['review']['user']['name'],end=" ")
    print(i['review']['rating'],end=" ")
    print(i['review']['review_text'])

12) Fetch the details of restaurant 736 AD using Zomato API.

13) Print the user rating, average cost for two, cuisines and address of 736 AD.

In [16]:
import requests as req
import json
val={'res_id':'18241524'}
res=req.get('https://developers.zomato.com/api/v2.1/restaurant',params=val,
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
#data
print(data['user_rating']['aggregate_rating'])
print(data['average_cost_for_two'])
print(data['cuisines'])
print(data['location']['address'])

4.3
1400
North Indian, Continental, Italian, Chinese
G-15/B, Vijay Nagar, New Delhi


14) Find out whether online table reservation is supported in 736 AD or not using Zomato API.

15) Print 'yes' if online table reservation is supported else 'no'

In [22]:
import requests as req
import json
val={'res_id':'18241524'}
res=req.get('https://developers.zomato.com/api/v2.1/restaurant',params=val,
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
if data['is_table_reservation_supported']==0:
    print('no')
else:
    print('yes')

no


16) Fetch the latest review of restaurant 736 AD using Zomato API

17) Print the name of user ,user rating and review text.

In [25]:
import requests as req
import json
val={'res_id':'18241524'}
res=req.get('https://developers.zomato.com/api/v2.1/reviews',params=val,
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
data
for i in data['user_reviews']:
    print(i['review']['user']['name'],end=" ")
    print(i['review']['rating'],end=" ")
    print(i['review']['review_text'])

Palak Sharma 4 
Ankita Sharma | Geographyedible 😎✌️ 5 All time favourite place ♥️💝
Lavanya Kv 4 
Foodieszonee 4 It’s a very nice place to hangout and has a very nice ambience, great dj and the food was very good must try white sauce pasta.  Good and fast service also reasonable price.
Prerana Verma 5 


18) Fetch the cuisine_id of of cuisine ‘BBQ’ using Zomato API.

19) Print the cuisine_id

In [32]:
import requests as req
import json
val={'city_id':'1'}
res=req.get('https://developers.zomato.com/api/v2.1/cuisines',params=val,
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
for i in data['cuisines']:
    if i['cuisine']['cuisine_name']=='BBQ':
        print(i['cuisine']['cuisine_id'])


193


20) Fetch the cuisine_id of of cuisine ‘BBQ’ using Zomato API.

21) Print the cuisine_id

In [56]:
import requests as req
import json
val={'lat':'28.697661','lon':'77.140626','cuisines':'193','sort':'real_distance','order':'desc','count':10}
res=req.get('https://developers.zomato.com/api/v2.1/search',params=val,
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
data

for i in data['restaurants']:
    print(i['restaurant']['name'],end=" ")
    rat=i['restaurant']['user_rating']
    print(rat['aggregate_rating'],end=" ")
    print(i['restaurant']['id'],end=" ")
    print(i['restaurant']['location']['locality'])

    


Themis Barbecue House 4.5 18204820 Netaji Subhash Place
Barbeque Nation 4.1 9561 Netaji Subhash Place
New Bombay Bar-Be-Que 3.6 5303 Prashant Vihar
Amazings Kitchen 0 19370424 Prashant Vihar
UBQ by Barbeque Nation 3.7 19082741 Lawrence Road
Moet's Barbeque 0 19238917 Alipur
UBQ by Barbeque Nation 3.8 18889760 Rohini
Kays Bar-Be-Que 3.8 3079 Ashok Vihar Phase 1
7teen BBQ Bistro Bnq 3.8 18595610 Moti Nagar
Pishori Chicken 0 19061375 Kirti Nagar


22) Fetch the latest review of the restaurant serving cuisine ‘BBQ’ which is most nearest to Coding ninjas?

Note : Due to limitation of basic API plan it, can only fetch 5 latest review

In [58]:
import requests as req
import json
val={'res_id':'18204820'}
res=req.get('https://developers.zomato.com/api/v2.1/reviews',params=val,
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
data
for i in data['user_reviews']:
    print(i['review']['user']['name'],end=" ")
    print(i['review']['rating'],end=" ")
    print(i['review']['review_text'])

ANNANT VASHIST 5 I celebrated my bday in Themis barbecue . I enjoyed a lot and the staff was really good nd the food is very delicious .
Thatcravingfoodie 4 Visited them last month. On entering, you realise that the indoor seating area is not that spacious. I wanted to be seated outdoors, but then it's only allowed for dinner time. I wasn't very taken by the ambiance.  Coming to the food, I opted for the non-vegetarian buffet option. For drinks, I chose a litchi based mocktail. You could also go for beer or whisky which I think I should have chosen too, now that I think of it. But there's no use crying over spilled milk now. So coming back to my litchi mocktail, let me say it was awesome. Tart and sweet and zingy. Was a pleasant accompaniment to the starters. I really liked the chilly fish, chicken tikka, crispy corn. Cajun potatoes lacked that kick and mutton seekh was really dry. The paneer tikka too wasn't flavourful enough. And the pizza base was very tough.  The main course was so

23) Print the name of user ,user rating and review text.

Note : Due to limitation of basic API plan it, can only fetch 5 latest review

In [61]:
import requests as req
import json
val={'lat':'28.697661','lon':'77.140626','cuisines':'193','sort':'rating','order':'desc','count':10,'radius':'4'}
res=req.get('https://developers.zomato.com/api/v2.1/search',params=val,
            headers={'user-key':'026f719e424a24871d19aaa0b9d4da47'})
data=res.json()
data

for i in data['restaurants']:
    print(i['restaurant']['name'],end=" ")
    rat=i['restaurant']['user_rating']
    print(rat['aggregate_rating'],end=" ")
    print(i['restaurant']['id'],end=" ")
    print(i['restaurant']['location']['locality'])

AB's Absolute Barbecues 4.9 18269368 Centurion Star Tower, Deira City Centre Area
Gary's Barbecue 4.9 17522122 China Grove
The Big Barbeque 4.9 19282432 Old Mahabalipuram Road (OMR)
WAKI Japanese BBQ Dining 4.9 18615392 Thamrin
Akins Bar-B-Q & Grill 4.9 17350006 Bell
Oakwood Cafe 4.9 17334212 Dalton
Biergarten 4.9 18313531 Whitefield
Dinosaur Bar-B-Que 4.9 17643027 Syracuse
Sparky's Burgers Espresso and BBQ 4.9 17535416 Hatch
Barbeque Nation 4.9 18734096 Cantonment
